client/client_test.go

```diff
@@ -98,3 +102,316 @@ func TestRedactPasswordArgs(t *testing.T) {
+func TestV2Forbidden(t *testing.T) { // Our test server.
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "text/plain")
+		w.WriteHeader(http.StatusForbidden)
+		w.Write([]byte(`Access forbidden`))
+	}))
+	defer ts.Close()
+
+	gsClient, err := NewV2(&Configuration{Endpoint: ts.URL})
+	if err != nil {
+		t.Error(err)
+	}
+
+	response, err := gsClient.CreateAuthToken("email", "password", nil)
+	if err == nil {
+		t.Error("Expected error, got nil")
+	}
+	if response != nil {
+		t.Error("Expected nil response")
+	}
+
+	clientAPIError, ok := err.(*clienterror.APIError)
+	if !ok {
+		t.Error("Type assertion err.(*clienterror.APIError) not successfull")
+	}
+
+	if clientAPIError.HTTPStatusCode != http.StatusForbidden {
+		t.Error("Expected HTTP status 403, got", clientAPIError.HTTPStatusCode)
+	}
+}
+
+// TestV2Unauthorized tests out how the latest client gives access to
+// HTTP error details for a 401 error
+func TestV2Unauthorized(t *testing.T) { // Our test server.
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusUnauthorized)
+		w.Write([]byte(`{"code": "PERMISSION_DENIED", "message": "Not authorized"}`))
+	}))
+	defer ts.Close()
+
+	gsClient, err := NewV2(&Configuration{Endpoint: ts.URL})
+	if err != nil {
+		t.Error(err)
+	}
+
+	_, err = gsClient.DeleteAuthToken("foo", nil)
+	if err == nil {
+		t.Error("Expected error, got nil")
+	}
+
+	t.Logf("err: %#v", err)
+
+	clientAPIError, ok := err.(*clienterror.APIError)
+	if !ok {
+		t.Error("Type assertion err.(*clienterror.APIError) not successfull")
+	}
+
+	if clientAPIError.HTTPStatusCode != http.StatusUnauthorized {
+		t.Error("Expected HTTP status 401, got", clientAPIError.HTTPStatusCode)
+	}
+}
+
+// TestV2AuxiliaryParams checks whether the client carries through our auxiliary
+// parameters
+func TestV2AuxiliaryParams(t *testing.T) { // Our test server.
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		if r.Header.Get("X-Request-ID") != "request-id" {
+			t.Error("Header X-Request-ID not available")
+		}
+		if r.Header.Get("X-Giant-Swarm-CmdLine") != "command-line" {
+			t.Error("Header X-Giant-Swarm-CmdLine not available")
+		}
+		if r.Header.Get("X-Giant-Swarm-Activity") != "activity-name" {
+			t.Error("Header X-Giant-Swarm-Activity not available")
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(`{"foo": "bar"}`))
+	}))
+	defer ts.Close()
+
+	config := &Configuration{
+		Endpoint: ts.URL,
+	}
+
+	gsClient, err := NewV2(config)
+	if err != nil {
+		t.Error(err)
+	}
+
+	ap := gsClient.DefaultAuxiliaryParams()
+	ap.RequestID = "request-id"
+	ap.CommandLine = "command-line"
+	ap.ActivityName = "activity-name"
+
+	_, _ = gsClient.CreateAuthToken("foo", "bar", ap)
```

Comment:
Just make it `gsClient.CreateAuthToken("foo", "bar", ap)`

Reply:
changed